internal/services/synapse/synapse_workspace_data_source.go

```diff
@@ -71,7 +71,11 @@ func dataSourceSynapseWorkspaceRead(d *pluginsdk.ResourceData, meta interface{})
 	if props := resp.WorkspaceProperties; props != nil {
 		d.Set("connectivity_endpoints", utils.FlattenMapStringPtrString(props.ConnectivityEndpoints))
 	}
-	if err := d.Set("identity", flattenIdentity(resp.Identity)); err != nil {
+	flattenIdentities, err := flattenIdentity(resp.Identity)
+	if err != nil {
+		return fmt.Errorf("retrieving Managed Identities for %s: %+v", id, err)
```

Comment:
this isn't retrieving the managed identities, but since this is a wrapped error this can become:

```suggestion
		return err
```